src/Features/CSharp/Portable/UseRecursivePatterns/AnalyzedNodeNormalizer.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseRecursivePatterns
+{
+    using static AnalyzedNode;
+
+    internal static class AnalyzedNodeNormalizer
+    {
+        public static AnalyzedNode Normalize(AnalyzedNode root)
+        {
+            var variables = PooledDictionary<ISymbol, Evaluation?>.GetInstance();
+            var result = VisitNode(root);
+            variables.Free();
+            return result;
+
+            AnalyzedNode VisitNode(AnalyzedNode node)
```

Comment:
can yuo make these sibling methods (or static local functions.  i have no clue what is being captured or not).